util/certificates.go

```diff
@@ -0,0 +1,79 @@
+package util
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+// ExtractCertificates takes the contents of the file at CF_INSTANCE_CERT, which typically are
+// comprised of two certificates. One is the identity certificate, and one is an intermediate
+// CA certificate which is crucial in linking the identity cert back to the configured root
+// certificate. It splits these two certificates apart, and identifies the certificate marked
+// as a CA as the intermediate cert, and the one not marked as a CA as the identity certificate.
+// It may error if the given file contents or certificates aren't as expected.
+func ExtractCertificates(cfInstanceCertContents string) (intermediateCert, identityCert *x509.Certificate, err error) {
```

Comment:
This method was added because through testing I learned that the instance certificate was actually _two_ certificates - the identity cert, and an intermediate CA that is required to tie the identity cert back to the root CA. This method simply parses those certificates into a more usable format.